### PR TITLE
Applicant save

### PR DESCRIPTION
**Pull request by ratneshraval**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- No tests added. 

#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/147277959

#### Brief feature description
- Controller action to save applicant by calling api. 

#### Does this PR block/effect other stories?
- No.

#### Do any commands needs to be run in order to run the project with this PR?
-

#### Code coverage percent diff as a result of this PR?
- 78%

#### JS errors diff as a result of this PR?
- 74

### CODE REVIEW

File: app/models/cals_base.rb [hunk 1]
```diff
@@ -8,4 +8,8 @@ def self.class_name_downcase_pluralized
   def self.api_resource_path
     name.demodulize.downcase.pluralize
   end
+
+  def self.parent_path
```
Comment (lauherkcambria) [inline app/models/cals_base.rb]: what is parent path and what does it represent?

File: app/models/concerns/RFA/applicant_api_protocol_provider.rb [hunk 2]
```diff
@@ -5,12 +5,10 @@ module Concerns::RFA::ApplicantApiProtocolProvider
   class_methods do
     def create(auth_header, parent_id, body)
       byebug
```
Comment (lauherkcambria) [inline app/models/concerns/RFA/applicant_api_protocol_provider.rb]: remove byebug

File: app/models/concerns/RFA/applicant_api_protocol_provider.rb [hunk 3]
```diff
@@ -5,12 +5,10 @@ module Concerns::RFA::ApplicantApiProtocolProvider
...
-      response = FaradayCals.post("/rfa-1a-forms/#{parent_id}/#{api_resource_path}", auth_header, body)
+      response = FaradayCals.post("/#{parent_path}/#{parent_id}/#{api_resource_path}", auth_header, body)
```
Comment (lauherkcambria) [inline app/models/concerns/RFA/applicant_api_protocol_provider.rb]: personally, this feels like over engineering, and part of the reason why it 'feels' like this architecture is more complicated than it is

File: app/models/helpers/model_helper_base.rb [hunk 4]
```diff
@@ -16,8 +16,12 @@ def all
     model_class.all(auth_header)
   end
 
+  # def create(url_path, body)
```
Comment (lauherkcambria) [inline app/models/helpers/model_helper_base.rb]: remove commented method

File: app/models/rfa/applicant.rb [hunk 5]
```diff
@@ -14,32 +14,28 @@ class RFA::Applicant < CalsBase
...
-  # desired children should move to application
-
   # has many license_history
 
-  # def other_names=(val)
-  #   @other_names = JSON.parse(val).map { |itm| AlternativeName.new(itm) }
-  # end
+  def other_names=(val)
+    @other_names = val.map { |itm| AlternativeName.new(itm) } if val.present?
+  end
```
Comment (lauherkcambria) [inline app/models/rfa/applicant.rb]: why have @ variable for other_names when class doesn't have other_names as an attr_accesor? what is the usage of these methods, and what is the passed in parameter 'val'? can it be more descriptive?

File: app/models/cals_base.rb (hunk 1 above)
Comment (ratneshraval) [inline app/models/cals_base.rb]: Some resources need a parent path like applicant needs /application/:id, so this code is an attempt plan for that.

File: app/models/concerns/RFA/applicant_api_protocol_provider.rb (hunk 2 above)
Comment (ratneshraval) [inline app/models/concerns/RFA/applicant_api_protocol_provider.rb]: Good catch. I'll remove it with some other updates.

File: app/models/helpers/model_helper_base.rb (hunk 4 above)
Comment (ratneshraval) [inline app/models/helpers/model_helper_base.rb]: Can we leave this in place. This is my attempt to move the create to base class.

File: app/models/rfa/applicant.rb (hunk 5 above)
Comment (ratneshraval) [inline app/models/rfa/applicant.rb]: @ other_names was added to the model to support names being part of the applicant object.

Comment (lauherkcambria) [inline app/models/rfa/applicant.rb]: in that case, i think :other_names needs to be added as an attr_accessor

Comment (ratneshraval) [inline app/models/rfa/applicant.rb]: I think it is already there.

**Pull request merged**